tests/NCronJob.Tests/TestHelper.cs

```diff
@@ -36,7 +40,25 @@ protected virtual void Dispose(bool disposing)
 
     protected async Task<bool> WaitForJobsOrTimeout(int jobRuns)
     {
-        using var timeoutTcs = new CancellationTokenSource(100);
+        using var timeoutTcs = new CancellationTokenSource(TimeSpan.FromSeconds(2));
+        try
+        {
+            await Task.WhenAll(GetCompletionJobs(jobRuns, timeoutTcs.Token));
+            return true;
+        }
+        catch (OperationCanceledException)
+        {
+            return true;
```

Comment:
Shouldn't this be `false` as well - indicating that the CTS kicked in

Reply:
I'm not sure. Maybe I misunderstood the intent of the method. I thought that either waiting for a Job or a Timeout should result in success because of how it was being used and the name of the method. I could use some help with this.

Reply:
I properly named the function very poorly.

The idea is to check that n tasks are completely successful. Therefore if you have a timeout, something went wrong.
It is a safety net around the `ChannelWriter` so that the CI pipeline fails quite early and doesn't run indefentily.

Reply:
I did checkout your branch locally and reverted the `true` inside the timeout back to `false` and tests started to fail.

Reply:
The reason seems to be how the `FakeTimeProvider` operates. Inside the `CronScheduler`:
```csharp
await Task.Delay(delay, timeProvider, stopToken);
```

We never progress from here if the `FakeTimeProvider` is used but we do with the `ManualTimeProvider`.
Given that plus the whole retry-mechanism it seems that the current setup isn't ideal.

Reply:
The idea is (because we have Timers in place) to autoadvance the Timer by the `ManualTimeProvider`

Reply:
I reverted the WaitForJobsOrTimeout to return false on operation cancelled. I fixed all tests, just needed to advance the time prior to reading from the Channel. I added an overload to WaitForJobsOrTimeout that allows us to pass the advancement of time necessary to trigger the Job run.